fmlloader/src/main/java/net/minecraftforge/fml/loading/ModSorter.java

```diff
@@ -242,7 +242,8 @@ private static String formatDependencyError(IModInfo.ModVersion dependency, Map<
 
     private boolean modVersionNotContained(final IModInfo.ModVersion mv, final Map<String, ArtifactVersion> modVersions)
     {
-        return !(VersionSupportMatrix.testVersionSupportMatrix(mv.getVersionRange(), mv.getModId(), "mod", (modId, range) -> modVersions.containsKey(modId) &&
+        return VersionSupportMatrix.isEnabled()
+                && !(VersionSupportMatrix.testVersionSupportMatrix(mv.getVersionRange(), mv.getModId(), "mod", (modId, range) -> modVersions.containsKey(modId) &&
```

Comment:
This isn't the correct change to the logic.
Namely that the standardLookup is never called if the enabled flag is false.
Honestly that 'standard lookup' needs to be moved outside the testVersionSupportMatrix method.
And having the ENABLED flag be overrideVersion.isEmpty would make the code cleaner and function better when version numbers change.

Reply:
The `if (MavenVersionAdapter.createFromVersionSpec("[1.19.2]").containsVersion(version)` (checking the current version before adding entries) would be the way i would go.
Tho honestly, I would skip all that crap. And just do 
```java
var mcver = FMLLoader.versionInfo().mcVersion();
if ("1.19.2".equals(mcver))
```

Bypass all the version range stuff.